Simplenote/src/androidTest/java/com/automattic/simplenote/login/LoginTests.java

```diff
@@ -0,0 +1,496 @@
+import androidx.test.espresso.PerformException;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.assertion.ViewAssertions;
+import androidx.test.espresso.contrib.RecyclerViewActions;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.espresso.util.HumanReadables;
+import androidx.test.espresso.util.TreeIterables;
+import androidx.test.filters.LargeTest;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.Until;
+
+import com.automattic.simplenote.BuildConfig;
+import com.automattic.simplenote.NotesActivity;
+import com.automattic.simplenote.R;
+import com.google.android.material.textfield.TextInputLayout;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import tools.fastlane.screengrab.Screengrab;
+import tools.fastlane.screengrab.UiAutomatorScreenshotStrategy;
+import tools.fastlane.screengrab.cleanstatusbar.CleanStatusBar;
+import tools.fastlane.screengrab.locale.LocaleTestRule;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LoginTests {
+    private static final int LAUNCH_TIMEOUT = 5000;
+
+    private static final String NOTE_FOR_EDITOR_SHOT_TITLE = "Lemon Cake & Blueberry";
+    private static final String NOTE_FOR_INTERLINKING_SHOT_TITLE = "# Colors";
+    private static final String NOTE_FOR_INTERLINKED_NOTE_SHOT_TITLE = "Blueberry Recipes";
+
+    @ClassRule
+    public static final LocaleTestRule localeTestRule = new LocaleTestRule();
+
+    @Rule
+    public ActivityTestRule<NotesActivity> mActivityTestRule = new ActivityTestRule<>(NotesActivity.class);
+
+    @BeforeClass
+    public static void beforeAll() {
+        Screengrab.setDefaultScreenshotStrategy(new UiAutomatorScreenshotStrategy());
+
+        new CleanStatusBar()
+                .setBatteryLevel(100)
+                .setClock("1231")
+                .setWifiVisibility(SHOW)
+                .setBarsMode(TRANSPARENT)
+                .enable();
+    }
+
+    @Test
+    public void loginTest() throws InterruptedException {
+        // Pre-checks if the state is dirty
+        enterThenDisablePasscodeIfNeeded();
+        dismissVerifyEmailScreenIfNeeded();
+        logoutIfNeeded();
+
+        // Login
+        login();
+
+        // Waiting for the notes (to make sure the login was successful)
+        Thread.sleep(2000);
```

Comment:
While I'm not free of this sin myself 😅 , it's preferable to avoid hard-coded sleeps and wait (with a timeout) for something to appear on the screen instead, like `All Notes` navigation bar of the FAB button.

One of the issues with hardcoded sleeps, which I stumbled upon recently, is the variability of the required time for the sleep: the test can run perfectly on a high-end laptop, but when executed as a part of CI cycle, where we usually have VMs, the used amount of 😴 might not be enough, because the VM performance might be lower.

Reply:
Fully agreed, I'm going to change the logic here